From: Yang Yingliang <yangyingliang@huawei.com>
Subject: fail_function: fix wrong use of fei_attr_remove()
Date: Fri, 26 Aug 2022 15:33:37 +0800

If register_kprobe() fails, the new attr is not added to the list yet, so
it should call fei_attr_free() intstead.

Link: https://lkml.kernel.org/r/20220826073337.2085798-3-yangyingliang@huawei.com
Fixes: 4b1a29a7f542 ("error-injection: Support fault injection framework")
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Reviewed-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 kernel/fail_function.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/kernel/fail_function.c~fail_function-fix-wrong-use-of-fei_attr_remove
+++ a/kernel/fail_function.c
@@ -295,7 +295,7 @@ static ssize_t fei_write(struct file *fi
 
 	ret = register_kprobe(&attr->kp);
 	if (ret) {
-		fei_attr_remove(attr);
+		fei_attr_free(attr);
 		goto out;
 	}
 	fei_debugfs_add_attr(attr);
_
